app/workers/delete_all_stale_objects_worker.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class DeleteAllStaleObjectsWorker < ApplicationJob
+
+  queue_as :deletion
+
+  def perform(*classes_names)
+    classes_names.each do |class_name|
+      delete_all_stale_in_batches class_name.constantize
+    end
+  end
+
+  private
+
+  def delete_all_stale_in_batches(model, batch_size: 500)
+    sleep 1 until model.stale.limit(batch_size).delete_all < batch_size
+  end
```

Comment:
Why not?

```suggestion
  def delete_all_stale_in_batches(model, batch_size: 500)
    model.stale.in_batches(of: batch_size) do |batch|
      batch.delete_all
      sleep 1
    end
  end
```

Looks more reliable to me.

Reply:
It is more complicated in this way and adds no benefits.  `in_batches` would add order to the results which is not necessary for our use case. Also it would try to track where we left so next batch skips what we went over. It probably would use more queries for this. Because it wouldn't know that records were deleted and it doesn't need to.

I think I checked it out and my recollection is that I saw unnecessary overhead. But if you try it out and see it is better, let me know.

Reply:
Delete all and limit are perfectly standard. Batches are when you would need to do something with the data. Query is slow anyway so adding more queries and redundant ordering can only make the operation more fragile. I have a pretty strong conviction not to change this.